src/main/java/com/minecolonies/core/entity/ai/workers/production/EntityAIQuarrier.java

```diff
@@ -669,35 +670,29 @@ public boolean walkToConstructionSite(final BlockPos currentBlock)
     {
         if (workFrom == null)
         {
-            workFrom = findRandomPositionToWalkTo(5, currentBlock);
-            if (workFrom == null && pathBackupFactor > 10)
+            if (EntityNavigationUtils.walkToRandomPosWithin(worker, 5, 1.0, job.findQuarry().getCorners()))
```

Comment:
This was supposed to get close to currentBlock and not take a fully random pos, no?

Reply:
ye should be around current